packages/datadog-plugin-aws-sdk/src/services/bedrockruntime.js

```diff
@@ -0,0 +1,254 @@
+    if (!enabledOperations.includes(operation)) {
+      return false
+    }
+
+    return super.isEnabled(request)
+  }
+
+  generateTags (params, operation, response) {
+    let tags = {}
+    let modelName = ''
+    let modelProvider = ''
+    const modelMeta = params.modelId.split('.')
+    if (modelMeta.length === 2) {
+      [modelProvider, modelName] = modelMeta
+      modelProvider = modelProvider.toUpperCase()
+    } else {
+      [, modelProvider, modelName] = modelMeta
+      modelProvider = modelProvider.toUpperCase()
+    }
+
+    const shouldSetChoiceIds = modelProvider === COHERE && !modelName.includes('embed')
+
+    const requestParams = extractRequestParams(params, modelProvider)
+    const textAndResponseReasons = extractTextAndResponseReason(response, modelProvider, modelName, shouldSetChoiceIds)
+
+    tags = buildTagsFromParams(requestParams, textAndResponseReasons, modelProvider, modelName, operation)
+
+    return tags
+  }
+}
+
+function extractRequestParams (params, provider) {
+  const requestBody = JSON.parse(params.body)
+  const modelId = params.modelId
+
+  if (provider === AI21) {
+    const temperature = requestBody.temperature || ''
+    const topP = requestBody.top_p || ''
+    const maxTokens = requestBody.max_tokens || ''
+    const stopSequences = requestBody.stop_sequences || []
+    let prompt = requestBody.prompt
+
+    if (modelId.includes('jamba')) {
+      for (const message of requestBody.messages) {
+        if (message.role === 'user') {
+          prompt = message.content // Return the content of the most recent user message
+        }
+      }
+    }
+    return {
+      prompt,
+      temperature,
+      top_p: topP,
+      max_tokens: maxTokens,
+      stop_sequences: stopSequences
+    }
+  } else if (provider === AMAZON && modelId.includes('embed')) {
+    return { prompt: requestBody.inputText }
+  } else if (provider === AMAZON) {
+    const textGenerationConfig = requestBody.textGenerationConfig || {}
+    return {
+      prompt: requestBody.inputText,
+      temperature: textGenerationConfig.temperature || '',
+      top_p: textGenerationConfig.topP || '',
+      max_tokens: textGenerationConfig.maxTokenCount || '',
+      stop_sequences: textGenerationConfig.stopSequences || []
+    }
+  } else if (provider === ANTHROPIC) {
+    const prompt = requestBody.prompt || ''
+    const messages = requestBody.messages || ''
+    return {
+      prompt: prompt || messages,
+      temperature: requestBody.temperature || '',
+      top_p: requestBody.top_p || '',
+      top_k: requestBody.top_k || '',
+      max_tokens: requestBody.max_tokens_to_sample || '',
+      stop_sequences: requestBody.stop_sequences || []
+    }
+  } else if (provider === COHERE && modelId.includes('embed')) {
+    return {
+      prompt: requestBody.texts,
+      input_type: requestBody.input_type || '',
+      truncate: requestBody.truncate || ''
+    }
+  } else if (provider === COHERE) {
+    return {
```

Comment:
Up to you but this common object shape might make for a good class, maybe something like `RequestParams` based on the function name.